api/insight/apiroutes.go

```diff
@@ -565,31 +568,32 @@ func (iapi *InsightApi) getTransactions(w http.ResponseWriter, r *http.Request)
 			return
 		}
 
-		txsOld := []*dcrjson.TxRawResult{}
-		txcount := len(blkTrans.RawTx) + len(blkTrans.RawSTx)
-		// Merge tx and stx together and limit result to 10 max
-		count := 0
-		for i := range blkTrans.RawTx {
-			txsOld = append(txsOld, &blkTrans.RawTx[i])
-			count++
-			if count > 10 {
-				break
-			}
+		// Sorting is not necessary since all of these transactions are in the
+		// same block, but put the stake transactions first.
+		mergedTxns := append(blkTrans.RawSTx, blkTrans.RawTx...)
+		txcount := len(mergedTxns)
+
+		pagesTotal := txcount / txPageSize
+		if txcount%txPageSize != 0 {
+			pagesTotal++
 		}
-		if count < 10 {
-			for i := range blkTrans.RawSTx {
-				txsOld = append(txsOld, &blkTrans.RawSTx[i])
-				count++
-				if count > 10 {
-					break
-				}
-			}
+
+		// Go to the last page if pageNum is greater than the number of pages.
+		if pageNum > pagesTotal {
+			pageNum = pagesTotal
+		}
+
+		// Grab the transactions for the given page (1-based index).
+		skipTxns := (pageNum - 1) * txPageSize
+		txsOld := []*dcrjson.TxRawResult{}
+		for i := skipTxns; i < txcount && i < txPageSize+skipTxns; i++ {
+			txsOld = append(txsOld, &mergedTxns[i])
```

Comment:
@buck54321 If you have a minute, could you review my page number logic in the 20 or so lines above?

Reply:
Any time in the next few days that is.  No hurry, just prioritizing 5.0.1 patch commits.